yt_dlp/extractor/common.py

```diff
@@ -860,8 +862,14 @@ def _request_webpage(self, url_or_request, video_id, note=None, errnote=None, fa
             headers = (headers or {}).copy()
             headers.setdefault('X-Forwarded-For', self._x_forwarded_for_ip)
 
+        extensions = None
+        if impersonate is not None:
+            target = ImpersonateTarget.from_str(':' if impersonate is True else impersonate)
+            if force_impersonate or self._downloader._impersonate_target_available(target):
+                extensions = {'impersonate': target}
```

Comment:
Multiple impersonate with priority is trivial to support if needed

```suggestion
        impersonate = tuple(map(ImpersonateTarget.from_str,
                                [':'] if impersonate is True else impersonate))
        extensions = {'impersonate': next(
            filter(self._downloader._impersonate_target_available, impersonate),
            impersonate[0] if force_impersonate and impersonate else None)}
```

Reply:
I expanded/improved upon this idea:

1. because if `extensions['impersonate'] == None`, then core code throws (i.e. if the key is present at all, then the value must be an instance of `ImpersonateTarget`)

2. to add a more specific warning when we aren't requiring impersonation but no targets are available

3. to pass an extractor-specific error message instead of using the generic core error message when we are requiring impersonation but no targets are available

4. to allow an `ImpersonateTarget` to be passed as the `impersonate` arg, per cole's suggestion below